### PR TITLE
srt(Section) quotes string values

### PR DESCRIPTION
**Pull request by yousifd**
Issue #972

### CODE REVIEW

**Comment by yousifd** [comment]
So, the tests are failing because I am modifying the output of the __str__ function from Section. Should I update the tests or is this going to be looked over by someone more familiar with the software?

**Comment by sils1297** [comment]
@yousifd thanks for your help! Adjusting the tests should be part of the commit. (Use --amend to change it)

I don't have time right now but I'll take a look at this first thing in the morning. You can ask us any questions on https://gitter.im/coala-analyzer/coala , we're happy to help, especially newcomers!

**Comment by Makman2** [comment]
@yousifd thanks for help!
what I had in mind is to use the `repr` function that wraps values correctly inside quotes, either `'` or `"` depending what the content is and even escapes those chars and other special sequences (like `\n`).
Could you try that? :)

**Comment by Makman2** [comment]
@yousifd If you have questions feel free to write here or at [gitter](https://gitter.im/coala-analyzer/coala), we are ~~evil~~ friendly guys :P

**Comment by sils1297** [comment]
> what I had in mind is to use the `repr` function that wraps values correctly inside quotes, either `'` or `"` depending what the content is, and even escapes those chars and other special sequences (like `\n`).
Could you try that? :)

@Makman2 ok now you're even confusing me a bit. We want strings to be wrapped in `'`s but not other objects, right? Ah you want to use `repr(str(...))` and `repr` will add the `'` encapsulation where appropriate, right?

**Comment by sils1297** [comment]
@Makman2 maybe you should clear that up in the bug :)

**Comment by Makman2** [comment]
@sils1297 exactly
and yeah I think you're right...

**Comment by Makman2** [comment]
@sils1297 I just see a DeprectationWarning inside the ConfigurationGatheringTest and this PR is not the cause. I file an issue^^

File: coalib/tests/settings/ConfigurationGatheringTest.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def test_gather_configuration(self):
             arg_list=['-S', "test=5", "-c", "some_bad_filename"])
 
         self.assertEqual(str(sections["default"]),
-                         "Default {config : some_bad_filename, test : 5}")
+                         "Default {config : \'some_bad_filename\', test : \'5\'}")
```
Comment (sils1297) [inline coalib/tests/settings/ConfigurationGatheringTest.py]: Codacy Issue found: [Line too long (82/80)](https://www.codacy.com/app/lasse/coala/file/1093749370/issues/source?bid=2303100&fileBranchId=2892270#l34)

**Comment by sils1297** [comment]
@yousifd let me summarize the tasks for this PR :)

- [ ] Please run `./coala` in the coala root directory to check for any problems. This command will show you what's wrong and sometimes be able to fix it automatically. (Well it's coala :))
- [ ] Instead of manually adding the quotes, you can just apply `repr` to the values, like pointed out by @Makman2 .
- [ ] Some tests still need adaption. You can run the tests with `./run_tests.py`, that should show you the pieces that need adaption.
- [ ] This is an aesthetic thing: can you squash all your commits into one? Just pull the latest master from our repository and then do a `git rebase coala/master` (assuming you call this remote `coala`), marking all but the first commits with `fixup` so they'll be squashed together. This is a bit advanced git stuff, we're happy to help you if you need assistance.

Ok that really looks much but it's just a bunch of only very minor things. :)

**Comment by yousifd** [comment]
I'll try my best during this week. Thanks for the suggestions.

**Comment by sils1297** [comment]
Oh I should probably mention that: we require each PR to be rebased on master before merge so please make sure you pull the latest master first, rebase on that and push up again. (With `--force`) This is a code quality thing plus we hate merge commits that clutter history :) a rebase is fast and cheap, nothing to worry about.

**Comment by Makman2** [comment]
@yousifd You still on it? Otherwise I would close this PR for now so it doesn't get stale forever^^

**Comment by yousifd** [comment]
@Makman2 Give me a couple of weeks. I am currently busy.

**Comment by Makman2** [comment]
@yousifd okay sure, just wanted to know :D

**Comment by sils1297** [comment]
@yousifd let us know if we can help you somehow!

**Comment by sils1297** [comment]
@yousifd you still on this?

**Comment by sils1297** [comment]
@yousifd I'll take this over then tomorrow if you don't mind. (You don't seem to.)

**Comment by Makman2** [comment]
#1137 takes over.

**Pull request closed without merging**